### ISSUE TITLE
🐛 Bug Report: Scaffolder Page contextMenu Props not correctly propagated

### ISSUE THREAD

**Issue by matdtr** [backstage/backstage#27448]
### 📜 Description

When configuring <ScaffolderPage> with contextMenu props as:

```
<ScaffolderPage contextMenu={{ editor: false }}/>
```

The template editor link correctly hidden but navigating to the action or tasks endpoint the 'editor' menu item is shown. 

### 👍 Expected behavior

The menu item should be hidden as in the `/create` endpoint:

<img width="1181" alt="Screenshot 2024-11-04 alle 12 58 25" src="https://github.com/user-attachments/assets/2138f398-ae3b-444c-8acf-7dcab97a6d71">

### 👎 Actual Behavior with Screenshots

The menu item is not hidden in the `/create/actions` or `/create/tasks` pages. 

<img width="1189" alt="Screenshot 2024-11-04 alle 12 58 41" src="https://github.com/user-attachments/assets/3884edf3-9bf6-4625-8f2a-e527bda78f31">

### 👟 Reproduction steps

1. Configure in `/packages/app/src/App.tsx` the following:

```
<ScaffolderPage contextMenu={{ editor: false }}/>
```

2. Navigate to /create/actions 

### 🖥️ Your Environment

```
yarn run v1.22.21
OS:   Darwin 23.6.0 - darwin/arm64
node: v18.19.0
yarn: 1.22.21
cli:  0.28.2 (installed)
backstage:  1.32.4

Dependencies:
  @backstage/app-defaults                                          1.5.12
  @backstage/backend-app-api                                       1.0.1
  @backstage/backend-common                                        0.24.1, 0.25.0
  @backstage/backend-defaults                                      0.5.2
  @backstage/backend-dev-utils                                     0.1.5
  @backstage/backend-openapi-utils                                 0.2.0
  @backstage/backend-plugin-api                                    0.8.1, 1.0.1
  @backstage/backend-tasks                                         0.6.1
  @backstage/backend-test-utils                                    1.0.2
  @backstage/catalog-client                                        1.7.1
  @backstage/catalog-model                                         1.7.0
  @backstage/cli-common                                            0.1.14
  @backstage/cli-node                                              0.2.9
  @backstage/cli                                                   0.28.2
  @backstage/config-loader                                         1.9.1
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.15.1
  @backstage/core-compat-api                                       0.2.5, 0.3.1
  @backstage/core-components                                       0.14.7, 0.15.1
  @backstage/core-plugin-api                                       1.10.0
  @backstage/dev-utils                                             1.1.2
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.10
  @backstage/frontend-app-api                                      0.10.0
  @backstage/frontend-defaults                                     0.1.1
  @backstage/frontend-plugin-api                                   0.6.5, 0.9.0
  @backstage/frontend-test-utils                                   0.2.1
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.2.0
  @backstage/integration                                           1.15.1
  @backstage/plugin-api-docs                                       0.11.11
  @backstage/plugin-app                                            0.1.1
  @backstage/plugin-auth-backend-module-atlassian-provider         0.3.1
  @backstage/plugin-auth-backend-module-auth0-provider             0.1.1
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.2.1
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.2.1
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.2.1
  @backstage/plugin-auth-backend-module-bitbucket-server-provider  0.1.1
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.3.1
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.3.1
  @backstage/plugin-auth-backend-module-github-provider            0.2.1
  @backstage/plugin-auth-backend-module-gitlab-provider            0.2.1
  @backstage/plugin-auth-backend-module-google-provider            0.2.1
  @backstage/plugin-auth-backend-module-guest-provider             0.2.1
  @backstage/plugin-auth-backend-module-microsoft-provider         0.2.1
  @backstage/plugin-auth-backend-module-oauth2-provider            0.3.1
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.2.1
  @backstage/plugin-auth-backend-module-oidc-provider              0.3.1
  @backstage/plugin-auth-backend-module-okta-provider              0.1.1
  @backstage/plugin-auth-backend-module-onelogin-provider          0.2.1
  @backstage/plugin-auth-backend                                   0.23.1
  @backstage/plugin-auth-node                                      0.5.3
  @backstage/plugin-auth-react                                     0.1.7
  @backstage/plugin-bitbucket-cloud-common                         0.2.24
  @backstage/plugin-catalog-backend-module-bitbucket-server        0.2.3
  @backstage/plugin-catalog-backend-module-msgraph                 0.6.3
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.2.1
  @backstage/plugin-catalog-backend                                1.27.1
  @backstage/plugin-catalog-common                                 1.1.0
  @backstage/plugin-catalog-graph                                  0.4.11
  @backstage/plugin-catalog-import                                 0.12.5
  @backstage/plugin-catalog-node                                   1.13.1
  @backstage/plugin-catalog-react                                  1.14.0
  @backstage/plugin-catalog                                        1.24.0
  @backstage/plugin-events-node                                    0.4.3
  @backstage/plugin-home-react                                     0.1.18
  @backstage/plugin-home                                           0.8.0
  @backstage/plugin-kubernetes-backend                             0.18.7
  @backstage/plugin-kubernetes-common                              0.8.3
  @backstage/plugin-kubernetes-node                                0.1.20
  @backstage/plugin-kubernetes-react                               0.4.4
  @backstage/plugin-kubernetes                                     0.11.16
  @backstage/plugin-org                                            0.6.31
  @backstage/plugin-permission-backend                             0.5.50
  @backstage/plugin-permission-common                              0.7.13, 0.8.1
  @backstage/plugin-permission-node                                0.8.4
  @backstage/plugin-permission-react                               0.4.27
  @backstage/plugin-proxy-backend                                  0.5.7
  @backstage/plugin-scaffolder-backend-module-azure                0.2.1
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.2.1
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.2.1
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.3.1
  @backstage/plugin-scaffolder-backend-module-cookiecutter         0.3.2
  @backstage/plugin-scaffolder-backend-module-gerrit               0.2.1
  @backstage/plugin-scaffolder-backend-module-gitea                0.2.1
  @backstage/plugin-scaffolder-backend-module-github               0.5.1
  @backstage/plugin-scaffolder-backend-module-gitlab               0.6.0
  @backstage/plugin-scaffolder-backend                             1.26.2
  @backstage/plugin-scaffolder-common                              1.5.6
  @backstage/plugin-scaffolder-node                                0.5.0
  @backstage/plugin-scaffolder-react                               1.13.2
  @backstage/plugin-scaffolder                                     1.26.2
  @backstage/plugin-search-backend-module-catalog                  0.2.4
  @backstage/plugin-search-backend-module-pg                       0.5.37
  @backstage/plugin-search-backend-module-techdocs                 0.3.1
  @backstage/plugin-search-backend-node                            1.3.4
  @backstage/plugin-search-backend                                 1.6.1
  @backstage/plugin-search-common                                  1.2.14
  @backstage/plugin-search-react                                   1.8.1
  @backstage/plugin-search                                         1.4.18
  @backstage/plugin-signals-react                                  0.0.6
  @backstage/plugin-techdocs-backend                               1.11.1
  @backstage/plugin-techdocs-common                                0.1.0
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.16
  @backstage/plugin-techdocs-node                                  1.12.12
  @backstage/plugin-techdocs-react                                 1.2.9
  @backstage/plugin-techdocs                                       1.11.0
  @backstage/plugin-user-settings-common                           0.0.1
  @backstage/plugin-user-settings                                  0.8.14
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.7.0
  @backstage/theme                                                 0.5.5, 0.6.0
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.10
Done in 1.28s.

```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by benjdlambert** [comment]
@camilaibs @jhaals is this related to the changes we made recently to the dropdown menu? :pray:

**Comment by jescalada** [comment]
Hi @benjdlambert and @matdtr, I just investigated this issue and it seems that originally, the `ActionsPage` and `TemplateListPage` context menus were originally hardcoded to have all the links by default. I just made a PR to extend them to receive the `contextMenu` prop and render the links conditionally.

The screenshots are available in #27565. Let me know if this is what you were looking for!

**Comment by matdtr** [comment]
Awesome! As https://github.com/backstage/backstage/pull/27565 has been merged, i will close this.

Thanks @jescalada!!
